session/schema_amender.go

```diff
@@ -194,7 +194,7 @@ func colChangeAmendable(colAtStart *model.ColumnInfo, colAtCommit *model.ColumnI
 // collectModifyColAmendOps is used to check if there is only column size increasing change.Other column type changes
 // such as column change from nullable to not null or column type change are not supported by now.
 func (a *amendCollector) collectModifyColAmendOps(tblAtStart, tblAtCommit table.Table) ([]amendOp, error) {
```

Comment:
Is this constraint check  `there is only column size increasing change`  still hold for this check function `collectModifyColAmendOps` by now or do we need to enhance it?

Reply:
When modifying the column type (which requires reorg the data), the column ID is changed. It will return [the error](https://github.com/pingcap/tidb/pull/26269/files#diff-e95622ea815b0964b17af236b3d707269eacad4d728fcb52646f22fb33f34bdbR210)  directly.  So we needn't update `colChangeAmendable`. Besides, we use `CheckModifyTypeCompatible` in `colChangeAmendable` which also updated in `column type change(with reorg data)`.